File: pocketr-java.api/src/main/java/com/decrux/pocketr/api/entities/dtos/AccountBalanceSummaryDto.java
package com.decrux.pocketr.api.entities.dtos;

import java.util.UUID;

public record AccountBalanceSummaryDto(
    UUID accountId,
    String accountName,
    String accountType,
    String currency,
    long balanceMinor
) {
    public UUID getAccountId() {
        return accountId;
    }

    public String getAccountName() {
        return accountName;
    }

    public String getAccountType() {
        return accountType;
    }

    public String getCurrency() {
        return currency;
    }

    public long getBalanceMinor() {
        return balanceMinor;
    }
}

File: pocketr-api/src/main/kotlin/com/decrux/pocketr/api/entities/dtos/AccountBalanceSummaryDto.kt
package com.decrux.pocketr.api.entities.dtos

import java.util.UUID

data class AccountBalanceSummaryDto(
    val accountId: UUID,
    val accountName: String,
    val accountType: String,
    val currency: String,
    val balanceMinor: Long,
)
